soroban-env-host/src/budget.rs

```diff
@@ -481,87 +453,71 @@ impl Default for BudgetImpl {
-                CostType::CloneEvents => cpu.lin_param = 10,
-                CostType::HostObjAllocSlot => cpu.const_param = 1000,
-
+                CostType::WasmInsnExec => cpu.lin_param = 32, // TODO: needs to be tiered
+                CostType::WasmMemAlloc => cpu.const_param = 6000, // TODO: questionable
+                CostType::InvokeHostFunction => {
+                    cpu.const_param = 5455;
+                    cpu.lin_param = 702
+                }
+                CostType::VisitObject => (), // This is 0
+                CostType::ValXdrConv => cpu.lin_param = 255,
+                CostType::ValSer => cpu.lin_param = 77,
+                CostType::ValDeser => cpu.lin_param = 19,
                 CostType::ComputeSha256Hash => {
-                    cpu.const_param = 3000;
-                    cpu.lin_param = 50;
+                    cpu.const_param = 3110;
+                    cpu.lin_param = 33;
                 }
-
-                CostType::ComputeEd25519PubKey => cpu.const_param = 40_000,
+                CostType::ComputeEd25519PubKey => cpu.const_param = 26_000,
                 CostType::VerifyEd25519Sig => {
-                    cpu.const_param = 1000;
-                    cpu.lin_param = 35;
+                    cpu.const_param = 367_202;
+                    cpu.lin_param = 22;
                 }
-
-                CostType::MapNew => cpu.lin_param = 2000,
-                CostType::MapEntry => cpu.lin_param = 100,
-                CostType::VecNew => cpu.lin_param = 2000,
-                CostType::VecEntry => cpu.lin_param = 100,
-                CostType::ScVecFromHostVec => cpu.lin_param = 10,
-                CostType::ScMapFromHostMap => cpu.lin_param = 10,
-                CostType::ScVecToHostVec => cpu.lin_param = 300,
-                CostType::ScMapToHostMap => cpu.lin_param = 2500,
-                CostType::GuardFrame => cpu.lin_param = 10,
-
-                CostType::VmMemRead => cpu.lin_param = 10,
-                CostType::VmMemWrite => cpu.lin_param = 10,
-                CostType::VmInstantiation => cpu.const_param = 100_000,
-                CostType::InvokeVmFunction => cpu.const_param = 10_000,
-                CostType::BytesClone
-                | CostType::BytesDel
-                | CostType::BytesPush
-                | CostType::BytesPop
-                | CostType::BytesInsert
-                | CostType::BytesAppend => cpu.lin_param = 10,
-                CostType::BytesCmp => cpu.lin_param = 1,
-                CostType::ChargeBudget => cpu.const_param = 50,
-                CostType::EdwardsPointCurve25519ScalarMul => cpu.const_param = 10,
-                CostType::HostMemAlloc | CostType::HostMemCpy => cpu.lin_param = 1,
+                CostType::MapNew => cpu.const_param = 188,
+                CostType::MapEntry => cpu.lin_param = 72,
+                CostType::VecNew => cpu.lin_param = 192,
```

Comment:
Agreed with @dmkozh some of these seem .. intuitively wrong, like there is probably a confounding measurement error.

Reply:
Sorry, it was my mistake copying number into the wrong field (vec should be linear too). Good catch.

Reply:
It was not a measurement error.

Reply:
Depends if mutating. Immutable access, yes. Mutable access requires duplicating the whole container. They're immutable and the schemes that exist for amortizing costs by reusing substructure wind up costing more for the small objects we're expecting to dominate than just duplicating the whole structure.

Reply:
Can we split mutable and immutable access? (not in this PR, of course). Accessing a vector element in linear time feels just wrong.

Also still want to remind about vec and map new mismatch that's nothing to do with the entry ops.

Reply:
+1 to @graydon on mutable access. 
Both `Map` and `Vec` here correspond to immutable, metered containers: `MeteredVector` and `MeteredMap`. `MeteredMap` is implemented using `MeteredVector`, so their model characteristics should be identical. 
For `MapNew` and `VecNew`, the input here means number of new Maps and Vecs are created. Therefore it is linear. In our code the input is always 1 as we don't create multiple empty containers at once, but for the metering purpose, the cost model needs to have an input defined.

Reply:
@dmkozh `MapEntry` here think of as "the cost of accessing a single map entry, if we've already located it", thus the linear cost here is correct. 
This is different from a typical user-facing "map.get(key)" operation where you need to perform binary search to find the key first. If you look at `metered_map::get` operation, it first performs the `find` operation which charges `log(N) * MapEntry` so the overall cost for the map operation works out correctly.
The whole metering framework is built from the bottom up consisting of lower level "components". Here the `MapEntry` is the component and `map.get` is an operation built from it.

Reply:
To note, it is plausible a mutable entry's cost is higher than immutable entry (similar to VmMemWrite vs VmMemRead) in which case it may be reasonable to split out `MapEntry` into `MapMutEntry` and `MapImmutEntry`, but the "MapMutEntry" will just have a higher linear factor.

Reply:
> Also still want to remind about vec and map new mismatch that's nothing to do with the entry ops.

I've fixed it now (thanks!)

Reply:
I don't think we actually provide mutable access (at least not yet) and the linear cost here refers to the number of entries being accessed (not the map size). That's ok for now, but we probably should try to make this a bit less confusing and standardize semantics of the `input` parameter.